File: Strings/1222.js
while (true) {
  try {
    let line1 = lines.shift().split(' ');
    let n = Number(line1[0]);
    let l = Number(line1[1]);
    let c = Number(line1[2]);
    let texto = lines.shift().split(' ');

    let linha = "";
    let linhas = 1;
    for (let i = 0; i < texto.length; i++) {
      let palavra = texto[i];
      if (linha == "") {
        linha = palavra;
      } else {
        if ((linha + " " + palavra).length > c) {
          linhas++;
          linha = palavra;
        } else {
          linha += " " + palavra;
        }
      }
    }

    let resultado = Math.ceil(linhas/l);

    console.log(resultado);
  } catch (error) {
    break;
  }
}

File: Strings/1222.py
import math

while True:
    try:
        n, l, c = input().split()
        l = int(l)
        c = int(c)
        texto = input().split()
        
        linha = ""
        linhas = 1
        for palavra in texto:
            if linha == "":
                linha = palavra
            else:
                if len(linha + " " + palavra) > c:
                    linhas += 1
                    linha = palavra
                else:
                    linha += " " + palavra
                    
        resultado = math.ceil(linhas/l)
        
        print(resultado)
    except EOFError:
        break
